### ISSUE TITLE
External types appearing as trait associated types are not detected

### ISSUE THREAD

**Issue by connec** [awslabs/cargo-check-external-types#19]
Apologies if this is a known limitation/expected behaviour, but currently it seems that when an external is used as an associated type in a trait implementation it is not detected. A trivial test case:

```sh
cargo new --lib local
cd local
echo 'external = { path = "external" }' >> Cargo.toml
cat <<EOF > src/lib.rs 
use external::ExternalThing;

pub struct LocalThing;

impl TryFrom<()> for LocalThing {
    type Error = ExternalThing;

    fn try_from(_value: ()) -> Result<Self, Self::Error> {
        todo!()
    }
}
EOF

cargo new --lib external
cat <<EOF > external/src/lib.rs 
pub struct ExternalThing;
EOF
```

Running `cargo check-external-types` shows no external types:

```
⟩ cargo +nightly-2022-07-25 check-external-types --output-format markdown-table
Running rustdoc to produce json doc output...
Examining all public types...
| Crate | Type | Used In |
| ---   | ---  | ---     |
```

**Comment by jdisanti** [comment]
Thanks for reporting this! It definitely looks like a bug.

I did some debugging into it, and it's happening because of the `is_public` call in `Visitor::visit_item`. This call is erroneously saying the associated type is private, when really, it should reflect the publicness of the trait being implemented.

**Comment by jdisanti** [comment]
The fix for this was released in [v0.1.5](https://github.com/awslabs/cargo-check-external-types/releases/tag/v0.1.5).
